src/lib/dnssd/tests/TestActiveResolveAttempts.cpp

```diff
@@ -43,50 +41,50 @@ Optional<ActiveResolveAttempts::ScheduledAttempt> ScheduledBrowse(const Dnssd::D
     return Optional<ActiveResolveAttempts::ScheduledAttempt>::Value(ActiveResolveAttempts::ScheduledAttempt(filter, type, first));
 }
 
-void TestSinglePeerAddRemove(nlTestSuite * inSuite, void * inContext)
+TEST(TestActiveResolveAttempts, TestSinglePeerAddRemove)
 {
     System::Clock::Internal::MockClock mockClock;
     mdns::Minimal::ActiveResolveAttempts attempts(&mockClock);
 
     mockClock.AdvanceMonotonic(1234_ms32);
 
     // Starting up, no scheduled peers are expected
-    NL_TEST_ASSERT(inSuite, !attempts.GetTimeUntilNextExpectedResponse().HasValue());
-    NL_TEST_ASSERT(inSuite, !attempts.NextScheduled().HasValue());
+    EXPECT_FALSE(attempts.GetTimeUntilNextExpectedResponse().HasValue());
+    EXPECT_FALSE(attempts.NextScheduled().HasValue());
 
     // Adding a single peer should result in it being scheduled
 
     attempts.MarkPending(MakePeerId(1));
 
-    NL_TEST_ASSERT(inSuite, attempts.GetTimeUntilNextExpectedResponse() == Optional<Timeout>(0_ms32));
-    NL_TEST_ASSERT(inSuite, attempts.NextScheduled() == ScheduledPeer(1, true));
-    NL_TEST_ASSERT(inSuite, !attempts.NextScheduled().HasValue());
+    EXPECT_EQ(attempts.GetTimeUntilNextExpectedResponse(), Optional<Timeout>(0_ms32));
```

Comment:
Maybe it's worth moving to `std::optional` here too (as a part of `chip::Optional` -> `std::optional` transition)?

Reply:
I'd rather change this in a separate PR, to keep this one small.